packages/driver/src/multi-domain/cypress.ts

```diff
@@ -73,6 +76,24 @@ const setup = () => {
 
   specBridgeCommunicator.initialize(window)
 
+  // TODO Should state syncing be built into cy.state instead of being explicitly called?
+  specBridgeCommunicator.on('sync:state', async (state) => {
+    cy.state(state)
+  })
+
+  // Listen for window load events from the primary window to resolve page loads
+  specBridgeCommunicator.on('window:load', ({ url }) => {
+    cy.isStable(true, 'load')
+    // @ts-ignore
+    Cypress.emit('internal:window:load', { type: 'cross:domain', url })
+  })
+
+  // Forward url:changed Message to the primary domain to enable changing the url displayed in the AUT
+  // @ts-ignore
+  Cypress.on('url:changed', (url) => {
```

Comment:
Is there a better place to listen to and forward this message?

Reply:
I think it's fine here.